File: frontend/src/types/position.ts
export interface PositionItem {
  id: number
  tenant_id: string
  account_id: number
  symbol: string
  market: string
  quantity: number
  available_quantity: number
  average_cost: string
  last_price: string
  unrealized_pnl: string
  realized_pnl: string
}

File: backend/app/schemas/position.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class PositionRead(BaseModel):
    id: int
    tenant_id: str
    account_id: int
    symbol: str
    name: str | None = None
    market: str
    quantity: int
    available_quantity: int
    average_cost: Decimal
    last_price: Decimal
    unrealized_pnl: Decimal
    realized_pnl: Decimal
    stop_loss_price: Decimal | None
    take_profit_price: Decimal | None
    strategy_add_count: int
    exit_guard_status: str
    exit_trigger_reason: str | None
    exit_triggered_at: datetime | None

    model_config = ConfigDict(from_attributes=True)
